### DISCUSSION TITLE
Showing all tick labels in ordinal bar chart

### DISCUSSION THREAD

**Discussion by Owez** [f5/unovis#150] [Q&A] [answered]
Hi, I'm using unovis in my SvelteKit application because it's the best looking graphing library out there :)

I'm trying to format my timeseries data in stacked graphs so I've parsed my data into ticks going from 0 -> x with human readable times being stored in a seperate array to do ordinal labels ([case 2](https://unovis.dev/docs/guides/tips-and-tricks/#displaying-ordinal-values) of the guide). This works perfectly for when there's 3 bars on the screen, for example:

<img width="300" alt="image" src="https://user-images.githubusercontent.com/30081702/221926015-dcf366f3-6001-41de-97d8-d77ed1f867fd.png">

The problem is that half of the labels vanish when 4 or more items appear:

<img width="300" alt="image" src="https://user-images.githubusercontent.com/30081702/221926225-6be8cd6b-1c39-4883-90b7-62cd15fdb941.png">

I need these labels because otherwise it looks as if the times in between are exactly halfway, even though this could be completely inaccurate. How should I go about enabling labels for every stacked bar? Here's the code I'm running to get these results:

https://github.com/Owez/yark/blob/61870215de07131cf2346ec3cb7ad13926a4c6eb/yark-pages/src/components/video/Graph.svelte

**Comment by rokotyan** [comment, marked as answer]
Hi @Owez, thanks for choosing Unovis!

You can specify the exact tick values for `Axis` using the `tickValues` property. Here's the section in the docs about it: https://unovis.dev/docs/auxiliary/Axis/#set-ticks-explicitly. In your case `tickValues` should be equal to `graphData.map(d => d.x)`, if I understood your code correctly.

**Reply by Owez** [reply]
Aha thanks! Looks like setting the `numTicks={x}` where x is the length of my `dateTicks` array is what I needed for my usecase in the end.

I didn't see this labelling documentation before, it doesn't seem to show up if you type "label"/"labels" in the search (only "labelling" pops it up). Might be worth prioritising it in the search if the provider your using allows doing that

**Reply by rokotyan** [reply]
@Owez Great! The `numTicks` property is indeed helpful when you need to add more ticks, however it might not work in some cases because that number is just an approximation. So if you have any troubles with the number of axis ticks it in the future, just switch to using `tickValues` instead of `numTicks`.

P.S. Thanks for pointing out the issue with the doc search
